src/test/librust/sapling_wallet_tests.cpp

```diff
@@ -0,0 +1,1133 @@
+    auto noteMap = wallet.GetSaplingScriptPubKeyMan()->FindMySaplingNotes(wtx).first;
+    BOOST_CHECK_EQUAL(0, noteMap.size());
+
+    // Add spending key to wallet, so Sapling notes can be found
+    BOOST_CHECK(wallet.AddSaplingZKey(sk));
+    BOOST_CHECK(wallet.HaveSaplingSpendingKey(extfvk));
+    noteMap = wallet.GetSaplingScriptPubKeyMan()->FindMySaplingNotes(wtx).first;
+    BOOST_CHECK_EQUAL(2, noteMap.size());
+
+    // Revert to default
+    RegtestDeactivateSapling();
+}
+
+// Generate note A and spend to create note B, from which we spend to create two conflicting transactions
+BOOST_AUTO_TEST_CASE(GetConflictedSaplingNotes) {
+    auto consensusParams = RegtestActivateSapling();
+
+    CWallet& wallet = *pwalletMain;
+    LOCK2(cs_main, wallet.cs_wallet);
+    setupWallet(wallet);
+
+    // Generate Sapling address
+    auto sk = GetTestMasterSaplingSpendingKey();
+    auto expsk = sk.expsk;
+    auto extfvk = sk.ToXFVK();
+    auto ivk = extfvk.fvk.in_viewing_key();
+    auto pk = sk.DefaultAddress();
+
+    BOOST_CHECK(wallet.AddSaplingZKey(sk));
+    BOOST_CHECK(wallet.HaveSaplingSpendingKey(extfvk));
+
+    // Generate note A
+    libzcash::SaplingNote note(pk, 50000);
+    auto cm = note.cmu().get();
+    SaplingMerkleTree saplingTree;
+    saplingTree.append(cm);
+    auto anchor = saplingTree.root();
+    auto witness = saplingTree.witness();
+
+    // Generate tx to create output note B
+    auto builder = TransactionBuilder(consensusParams, 1);
+    builder.AddSaplingSpend(expsk, note, anchor, witness);
+    builder.AddSaplingOutput(extfvk.fvk.ovk, pk, 35000, {});
+    auto tx = builder.Build().GetTxOrThrow();
+    CWalletTx wtx {&wallet, tx};
+
+    // Fake-mine the transaction
+    BOOST_CHECK_EQUAL(0, chainActive.Height());
+    CBlock block;
+    block.hashPrevBlock = chainActive[0]->GetBlockHash();
+    block.vtx.emplace_back(MakeTransactionRef(wtx));
+    block.hashMerkleRoot = BlockMerkleRoot(block);
+    auto blockHash = block.GetHash();
+    CBlockIndex fakeIndex {block};
+    BlockMap::iterator mi = mapBlockIndex.emplace(blockHash, &fakeIndex).first;
+    fakeIndex.phashBlock = &((*mi).first);
+    chainActive.SetTip(&fakeIndex);
+    BOOST_CHECK(chainActive.Contains(&fakeIndex));
+    BOOST_CHECK_EQUAL(0, chainActive.Height());
+
+    // Simulate SyncTransaction which calls AddToWalletIfInvolvingMe
+    auto saplingNoteData = wallet.GetSaplingScriptPubKeyMan()->FindMySaplingNotes(wtx).first;
+    BOOST_CHECK(saplingNoteData.size() > 0);
+    wtx.SetSaplingNoteData(saplingNoteData);
+    wtx.SetMerkleBranch(mapBlockIndex[blockHash], 0);
+    BOOST_CHECK(wallet.LoadToWallet(wtx));
+
+    // Simulate receiving new block and ChainTip signal
+    wallet.IncrementNoteWitnesses(&fakeIndex, &block, saplingTree);
+    wallet.GetSaplingScriptPubKeyMan()->UpdateSaplingNullifierNoteMapForBlock(&block);
+
+    // Retrieve the updated wtx from wallet
+    uint256 hash = wtx.GetHash();
```

Comment:
(9cabc2ed2399221b23491c536254488dda44e60e) there's few more instances...
```suggestion
    const uint256& hash = wtx.GetHash();
```